src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.cs

```diff
@@ -491,6 +494,64 @@ public async Task forkchoiceUpdatedV1_should_work_with_zero_keccak_for_finalizat
             AssertExecutionStatusChanged(rpc, newHeadHash!, Keccak.Zero, startingHead);
         }
         
+        [Test]
+        public async Task forkchoiceUpdatedV1_should_update_finalized_block_hash()
+        {
+            using MergeTestBlockchain chain = await CreateBlockChain();
+            TestRpcBlockchain testRpc = await CreateTestRpc(chain);
+            IEngineRpcModule rpc = CreateEngineModule(chain);
+            Keccak startingHead = chain.BlockTree.HeadHash;
+            BlockRequestResult blockRequestResult = await SendNewBlockV1(rpc, chain);
+
+            Keccak newHeadHash = blockRequestResult.BlockHash;
+            ForkchoiceStateV1 forkchoiceStateV1 = new(newHeadHash!, startingHead, startingHead!);
+            ResultWrapper<ForkchoiceUpdatedV1Result> forkchoiceUpdatedResult =
+                await rpc.engine_forkchoiceUpdatedV1(forkchoiceStateV1, null);
+            forkchoiceUpdatedResult.Data.PayloadStatus.Status.Should().Be(PayloadStatus.Valid);
+            forkchoiceUpdatedResult.Data.PayloadId.Should().Be(null);
+            
+            Keccak? actualFinalizedHash = chain.BlockTree.FinalizedHash;
+            actualFinalizedHash.Should().NotBeNull();
+            actualFinalizedHash.Should().Be(startingHead);
+            
+            BlockForRpc blockForRpc = testRpc.EthRpcModule.eth_getBlockByNumber(BlockParameter.Finalized).Data;
+            blockForRpc.Should().NotBeNull();
+            actualFinalizedHash = blockForRpc.Hash;
+            actualFinalizedHash.Should().NotBeNull();
+            actualFinalizedHash.Should().Be(startingHead);
+            
+            AssertExecutionStatusChanged(rpc, newHeadHash!, startingHead, startingHead);
+        }
+        
+        [Test]
+        public async Task forkchoiceUpdatedV1_should_update_safe_block_hash()
```

Comment:
we need to write tests with BlockFinder or EthRpcModuleTests (even better)

Reply:
I already check in this test that rpc module returns the correct finalized/safe blocks. what else u suggest?